lib/developer_portal/app/controllers/developer_portal/admin/applications_controller.rb

```diff
@@ -113,10 +113,12 @@ def applications
   end
 
   def service
-    @service ||= if not site_account.multiservice?
-                   site_account.services.default
-                 elsif service_id = params[:service_id]
-                   site_account.services.find_by_id_or_system_name(service_id)
+    @service ||= if single_service?
+                   services.default
+                 elsif (service_id = params[:service_id])
+                   services.where(id: service_id).or(services.where(system_name: service_id)).first
+                 elsif current_account.services_can_create_app_on.count == 1
+                   current_account.services_can_create_app_on.first
```

Comment:
`Account#services_can_create_app_on` means only services that the buyer account already has at least one `ServiceContract` with. Meaning this condition does not cover for the case where the buyer has `ServiceContract` with one service and here tries to create a new application with an application plan of another service. Is this what we want here?

Sorry. I still don't understand this condition very well.

Reply:
All I can say is that this is the current behaviour 🤷‍♂️

Reply:
And what happens when multi-services is enabled, the buyer lands to the new application page without a `service_id` and the buyer has not `ServiceContract`? `service` will be `nil`, right?

Reply:
cc @thomasmaas

Reply:
If there are no subscriptions for the account, the "create application" button will be hidden.
https://github.com/3scale/porta/blob/df50136a2d268c873373bef227680153354f2a83/lib/developer_portal/app/views/developer_portal/applications/index.html.liquid#L42-L44

If the user manually goes to `applications/new`: `service` will be `nil` and they will be redirected to the `choose_service` screen, which will be empty.

Reply:
Fair enough. Let's move on and see how it goes. I guess "choosing the service" is the critical step that absolutely needs to happen upfront before hitting this point.

Thanks for explaining!